server/py/framework/db/sqldb/db.py

```diff
@@ -3262,41 +3247,34 @@ def _get_project_record(
         return project_record
 
     def verify_project_has_no_related_resources(self, session: Session, name: str):
-        artifacts = self._find_artifacts(session, project=name, ids="*")
-        self._verify_empty_list_of_project_related_resources(
-            name, artifacts, "artifacts"
-        )
-        logs = self._list_logs(session, name)
-        self._verify_empty_list_of_project_related_resources(name, logs, "logs")
-        runs = self._find_runs(session, None, name, []).all()
-        self._verify_empty_list_of_project_related_resources(name, runs, "runs")
-        notifications = []
+        # it is enough to sample few resources, we do not need to retrieve all resources really.
+        resource_limit = 5
+        for resource_name, resource_list_function, kwargs in [
+            ("runs", self.list_runs, dict(sort=False, return_as_run_structs=False)),
+            ("artifacts", self._find_artifacts, dict(ids="*")),
```

Comment:
Do these params matter?
`sort` definitely doesn't matter here, also `return_as_run_structs` doesn't matter as you don't care about the type of the returned resources only if the list is empty. and artifacts' `ids=*` is the same as `ids=None`.
Looks like you don't necessarily need any kwargs in this for loop, you only need resource name and method.